src/viam/components/camera/service.py

```diff
@@ -61,6 +66,31 @@ async def GetImage(self, stream: Stream[GetImageRequest, GetImageResponse]) -> N
         response.image = img_bytes
         await stream.send_message(response)
 
+    async def GetImages(self, stream: Stream[GetImagesRequest, GetImagesResponse]) -> None:
+        request = await stream.recv_message()
+        assert request is not None
+        name = request.name
+        camera = self.get_resource(name)
+
+        timeout = stream.deadline.time_remaining() if stream.deadline else None
+        images, ts = await camera.get_images(timeout=timeout, metadata=stream.metadata)
+        img_bytes_lst = []
+        for img in images:
+            try:
+                # encode all pil images as jpg as default
+                mimetype = CameraMimeType.JPEG
+                if isinstance(img, RawImage):
+                    mimetype, _ = CameraMimeType.from_lazy(img.mime_type)
+                fmt = mimetype.to_proto()
+                img_bytes = mimetype.encode_image(img)
+                img_bytes_lst.append(Image(source_name=name, format=fmt, image=img_bytes))
```

Comment:
the source_name might not be the camera name- it should be whatever string the server associated with the image.